promgen/static/js/promgen.vue.js

```diff
@@ -103,23 +92,47 @@ var app = new Vue({
                 });
 
         }
-
     },
     computed: {
         alertLabelsService: function () {
-            return this.filterBy(this.globalAlerts, 'service');
+            return new Set(this.globalAlerts
```

Comment:
I think better to define this process to a `method` and call it from computed property.
So maintenance will be easily.

Reply:
I previously wrote it as my own method, but I think that even though it's a few more lines in the computed properly itself, it's much more readable and more maintainable in this format

Reply:
Like this

```js
  methods: {
    filterByService: function() {
      return new Set(
        this.globalAlerts
          .filter(x => x.status.state == 'active')
          .filter(x => x.labels.service)
          .map(x => x.labels.service)
          .sort()
      )
    },
    filterByProject: function() {
      return new Set(
        this.globalAlerts
          .filter(x => x.status.state == 'active')
          .filter(x => x.labels.project)
          .map(x => x.labels.project)
          .sort()
      )
    },
    filterByAlertName: function() {
      return new Set(
        this.globalAlerts
          .filter(x => x.status.state == 'active')
          .filter(x => x.labels.alertname)
          .map(x => x.labels.alertname)
          .sort()
      )
    }
  },
  computed: {
    alertLabelsService: function() {
      return this.filterByService
    },
    alertLabelsProject: function() {
      return this.filterByProject
    },
    alertLabelsRule: function() {
      return this.filterByAlertName
    },
    silenceLabelsService: function() {
      return this.filterByService
    },
    silenceLabelsProject: function() {
      return this.filterByProject
    },
}
```

Reply:
I'm sorry. This code was wrong.